plugins/importexport/medra/MedraExportPlugin.inc.php

```diff
@@ -115,14 +130,62 @@ function depositXML($objects, $context, $filename) {
 				$this->saveRegisteredDoi($context, $object);
 			}
 		} else {
-			// Handle errors.
-			if (is_string($result)) {
-				$result = array(
-					array('plugins.importexport.common.register.error.mdsError', $result)
-				);
-			} else {
-				$result = false;
-			}
+			// Handle errors. There are validations before sending the request of submission to Crossref endpoint, and there is a need to throw a readable exception to the end user
+			//the exception are shown in a table as represented in the code below.
+		    if(!assert(PKPString::regexp_match('#<returnCode>success</returnCode>#', $result))){
+		        $doc = new DOMDocument();
+		        $doc->loadXML($result);
+		        $templateMgr = TemplateManager::getManager($this->_request);
+		        $numberError = '';
+		        
+		        if($doc->getElementsByTagName('statusCode')->item(0)->textContent == 'FAILED'){
+		            $numberError = $doc->getElementsByTagName('errorsNumber')->item(0)->textContent;
+		            $nodeList = $doc->getElementsByTagName('error');
+		            
+		            $headlines = array();
+		            
+		            foreach($nodeList as $node) {
+		                $headline = array();
+		                if($node->childNodes->length) {
+		                    foreach($node->childNodes as $i) {
+		                        $headline[$i->nodeName] = $i->nodeValue;
+		                    }
+		                }
+		                
+		                $headlines[] = $headline;
+		            }
+		            
+		            $templateMgr->assign(
+		                'headlines', $headlines
+		            );
+		            
+		        }
+		        
+		        $templateMgr->assign(
+		            'htmlspecialchars', htmlspecialchars($xml)
+		        );
+		        
+		        $templateMgr->assign(
+		            'numberError', $numberError
+		        );
+		        
+		        $templateMgr->display($this->getTemplateResource('crDepositErrors.tpl'));
+		        
+		    } else
+		        if (is_string($result)) {
+		            
+		            error_log($result);
```

Comment:
I think this is here for debugging purposes only, correct? If so, it should be removed.

Reply:
Done, i removed that line code. it was there for debugging purposes